src/main/java/duke/command/AddCommand.java

```diff
@@ -0,0 +1,146 @@
+
+    /**
+     * Represents the type of add command.
+     * TODO: to add a ToDo Task, DEADLINE: to add a Deadline Task, EVENT: to add an Event task.
+     */
+    public enum AddType {
+        TODO, DEADLINE, EVENT;
+    }
+
+    /**
+     * Constructs an AddCommand object.
+     *
+     * @param addType the type of adding command.
+     * @param input the user's input.
+     */
+    public AddCommand(AddType addType, String input) {
+        super(input);
+        this.addType = addType;
+    }
+
+    /**
+     * Executes the adding command accordingly.
+     *
+     * @param tasksList the tasks list of Duke.
+     * @param ui the ui of Duke.
+     * @param database the database of Duke.
+     * @throws DukeException if the user's input is incorrect.
+     */
+    public void execute(TaskList tasksList, Ui ui, DukeDatabase database) throws DukeException {
+        initialise(tasksList, ui, database);
+
+        if (AddType.TODO.equals(addType)) {
+            addToDo();
+        } else if (AddType.DEADLINE.equals(addType)) {
+            addDeadline();
+        } else if (AddType.EVENT.equals(addType)) {
+            addEvent();
+        } else {
+            throw new DukeException("Internal logic bug occurred!");
+        }
+    }
+
+    /**
+     * Adds to do entry to the taskList.
+     */
+    private void addToDo() throws DukeException {
+        String topic = input.substring(4).trim();
+
+        if ("".equals(topic)) {
+            throw new DukeException("The description of a todo cannot be empty.");
+        }
+
+        addTask(new ToDo(topic));
+    }
+
+    /**
+     * Adds deadline entry to the taskList.
+     */
+    private void addDeadline() throws DukeException {
+        String[] details = input.substring(8).trim().split("/by");
+
+        if (details.length == 1 || "".equals(details[0].trim()) || "".equals(details[1].trim())) {
+            throw new DukeException("The description and deadline of a deadline cannot be empty.");
+        }
+
+        try {
+            String topic = details[0].stripTrailing();
+            String deadlineUserInput = details[1].stripLeading();
+            String deadline = formatDateAndTime(deadlineUserInput);
+
+            addTask(new Deadline(topic, deadline));
+        } catch (ParseException e) {
+            ui.echo(DukeException.PREFIX + " The format of date and time is wrong!");
+        }
+    }
+
+    /**
+     * Adds event entry to the taskList.
+     */
+    private void addEvent() throws DukeException {
+        String[] details = input.substring(5).trim().split("/at");
+
+        if (details.length == 1 || "".equals(details[0].trim()) || "".equals(details[1].trim())) {
+            throw new DukeException("The description and date of an event cannot be empty.");
+        }
+
+        try {
+            String topic = details[0].stripTrailing();
+            String dateUserInput = details[1].stripLeading();
+            String date = formatDateAndTime(dateUserInput);
+
```

Comment:
An alternative to String manipulation would be to use Regular Expressions. You can refer to my fork in Duke.java to see how it could be used. As the complexity of the commands increase, writing equivalent String Manipulation code would become increasingly tedious.

In our use case, the commands are trivial, and would likely be faster than using Regex, however the tradeoff would be a little more refactoring in the future when commands become more complex.